lib/ret/media_resolver.ex

```diff
@@ -454,41 +454,60 @@ defmodule Ret.MediaResolver do
         _err -> [uri, nil]
       end
 
-    {:commit, uri |> resolved(meta)}
+    {:commit, resolved(uri, meta)}
   end
 
-  defp resolve_sketchfab_model(model_id, api_key, version \\ 1) do
-    cached_file_result =
-      CachedFile.fetch(
-        "sketchfab-#{model_id}-#{version}",
-        fn path ->
-          Statix.increment("ret.media_resolver.sketchfab.requests")
+  defp get_sketchfab_model_zip_url(%{model_id: model_id, api_key: api_key}) do
+    case "https://api.sketchfab.com/v3/models/#{model_id}/download"
+         |> retry_get_until_success([{"Authorization", "Token #{api_key}"}], 15_000, 15_000) do
+      :error ->
+        {:error, "Failed to get sketchfab metadata"}
+
+      response ->
+        case response |> Map.get(:body) |> Poison.decode() do
+          {:ok, json} ->
+            {:ok, Kernel.get_in(json, ["gltf", "url"])}
+
+          _ ->
+            {:error, "Failed to get sketchfab metadata"}
+        end
+    end
+  end
 
-          res =
-            "https://api.sketchfab.com/v3/models/#{model_id}/download"
-            |> retry_get_until_success([{"Authorization", "Token #{api_key}"}])
+  def download_sketchfab_model_to_path(%{model_id: model_id, api_key: api_key, path: path}) do
+    case get_sketchfab_model_zip_url(%{model_id: model_id, api_key: api_key}) do
+      {:ok, zip_url} ->
+        Download.from(zip_url, path: path)
+        {:ok, %{content_type: "model/gltf+zip"}}
 
-          case res do
-            :error ->
-              Statix.increment("ret.media_resolver.sketchfab.errors")
+      {:error, error} ->
+        {:error, error}
 
-              :error
+      _ ->
+        {:error, "Failed to get sketchfab url"}
+    end
+  end
 
-            res ->
-              Statix.increment("ret.media_resolver.sketchfab.ok")
+  defp resolve_sketchfab_model(model_id, api_key, version \\ 1) do
+    loader = fn path ->
+      Statix.increment("ret.media_resolver.sketchfab.requests")
 
-              zip_url =
-                res
-                |> Map.get(:body)
-                |> Poison.decode!()
-                |> Kernel.get_in(["gltf", "url"])
+      case download_sketchfab_model_to_path(%{model_id: model_id, api_key: api_key, version: version, path: path}) do
+        {:ok, metadata} ->
+          Statix.increment("ret.media_resolver.sketchfab.ok")
+          {:ok, metadata}
 
-              Download.from(zip_url, path: path)
+        {:error, _} ->
+          Statix.increment("ret.media_resolver.sketchfab.errors")
+          :error
+      end
+    end
 
-              {:ok, %{content_type: "model/gltf+zip"}}
-          end
-        end
-      )
+    cached_file_result =
+      CachedFile.fetch(%{
+        cache_key: "sketchfab-#{model_id}-#{version}",
+        loader: loader
+      })
```

Comment:
Personally, I don't find positional arguments confusing enough to have a burning desire to replace them everywhere in our reticulum and javascript code. It's not a strong opinion, so I'd rather not spend a disproportionate amount of time discussing it one way or the other. If we do decide to adopt the pattern for whatever reason, I think I'd like to understand the rules around when we do and don't use it (e.g. when exactly is it a performance concern?), do it all in one go, and implement some linting to prevent my muscle memory from regressing the pattern.

Reply:
I don't feel strongly about this either and don't want to suggest adopting something that would call for a bunch of refactoring for little, no, or possibly negative value. I've been curious if anyone would feel similarly about this but I'll save it for another time / project.